packages/smart-contracts/scripts/deploy-payments.ts

```diff
@@ -215,27 +228,38 @@ export async function deployAllPaymentContracts(
       nonceCondition: 1,
     });
 
+    // Batch 3
+    const nonceForBatch3 = 7;
+    await jumpToNonce(args, hre, nonceForBatch3);
+
+    const { address: ethFeeProxyAddress } = await runEasyDeployment({
+      contractName: 'EthereumFeeProxy',
+      artifact: ethereumFeeProxyArtifact,
+      nonceCondition: nonceForBatch3,
+    });
+
+    // Batch 4
+    const nonceForBatch4 = 10;
+    await jumpToNonce(args, hre, nonceForBatch4);
+
     const {
       instance: chainlinkInstance,
       address: chainlinkInstanceAddress,
     } = await runEasyDeployment({
       contractName: 'ChainlinkConversionPath',
+      constructorArguments: [nativeTokenHash],
       artifact: chainlinkConversionPathArtifact,
-      nonceCondition: 2,
+      nonceCondition: nonceForBatch4,
     });
 
-    // Batch 2
-    await runDeploymentBatch_2(chainlinkInstance, erc20FeeProxyAddress);
+    await runDeploymentBatch_4(chainlinkInstanceAddress, ethFeeProxyAddress);
 
-    // Batch 3
-    const { address: ethFeeProxyAddress } = await runEasyDeployment({
-      contractName: 'EthereumFeeProxy',
-      artifact: ethereumFeeProxyArtifact,
-      nonceCondition: 7,
-    });
+    if (!args.simulate) {
+      throw new Error('Missing implementation new version of ERC20 Conversion Proxy');
+    }
 
-    // Batch 4
-    await runDeploymentBatch_4(chainlinkInstanceAddress, ethFeeProxyAddress);
+    // Batch 2
+    await runDeploymentBatch_2(chainlinkInstance, erc20FeeProxyAddress);
```

Comment:
Maybe I missed something, but it seems that when running :
```javascript
await runDeploymentBatch_2(chainlinkInstance, erc20FeeProxyAddress);
```
You should already have a nonce greater than 10 as it is the nonce condition to deploy the ChainlinkConversionPath contract. However there are several hardcoded nonce conditions lower than 10 in this function.

Reply:
That's why I added the error thrown just above, line 258, I'll fix that later but now I want to solve the incident quickly.